test/Microsoft.VisualStudio.Editor.Razor.Test.Common/StringTextSnapshot.cs

```diff
@@ -4,15 +4,32 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using Microsoft.VisualStudio.Utilities;
 
 namespace Microsoft.VisualStudio.Text
 {
     public class StringTextSnapshot : ITextSnapshot
     {
+        private readonly List<ITextSnapshotLine> _lines;
+
         public StringTextSnapshot(string content)
         {
             Content = content;
+            _lines = new List<ITextSnapshotLine>();
+
+            var start = 0;
+            var delimiterIndex = 0;
+            while (delimiterIndex != -1)
+            {
+                delimiterIndex = Content.IndexOf(Environment.NewLine, start);
+                var nextLineStartIndex = delimiterIndex != -1 ? delimiterIndex + Environment.NewLine.Length : Content.Length;
```

Comment:
This isn't right, we need to be able to work with all line break types in all environements